### PR TITLE
Meta: QtCreator setup and formatting; and linting

### PR DESCRIPTION
**Pull request by BenWiederhake**
The next one won't be a mixed bag ^^

Thanks to @vkoskiv for the great write-up!

A few months ago I fell into the trap of not naming the Qt project `serenity`, but rather `SerenityOS`, and I want to spare future users from fixing such a blunder. Also, this documents auto-format, and kits.

While watching the video in which you explained IPC I noticed that we have IDs as a fail-safe mechanism to prevent wrong connections, but nowhere do we actually ensure that these IDs are unique. Well, now we do.

Here's how it looks like when the check fails:
```
This is the full list of all endpoints:
./Services/WindowServer/WindowServer.ipc:endpoint WindowServer = 2
./Services/WindowServer/WindowClient.ipc:endpoint WindowClient = 4
./Services/ProtocolServer/ProtocolServer.ipc:endpoint ProtocolServer = 9
./Services/ProtocolServer/ProtocolClient.ipc:endpoint ProtocolClient = 13
./Services/AudioServer/AudioClient.ipc:endpoint AudioClient = 82
./Services/AudioServer/AudioServer.ipc:endpoint AudioServer = 85
./Services/AudioServer/StealthServer.ipc:endpoint EvilServer = 85
./Services/WebContent/WebContentServer.ipc:endpoint WebContentServer = 89
./Services/WebContent/WebContentClient.ipc:endpoint WebContentClient = 90
./Services/NotificationServer/NotificationClient.ipc:endpoint NotificationClient = 92
./Services/NotificationServer/NotificationServer.ipc:endpoint NotificationServer = 95
./Services/LaunchServer/LaunchServer.ipc:endpoint LaunchServer = 101
./Services/LaunchServer/LaunchClient.ipc:endpoint LaunchClient = 102
./Services/Clipboard/ClipboardServer.ipc:endpoint ClipboardServer = 802
./Services/Clipboard/ClipboardClient.ipc:endpoint ClipboardClient = 804
./Services/ImageDecoder/ImageDecoderServer.ipc:endpoint ImageDecoderServer = 7001
./Services/ImageDecoder/ImageDecoderClient.ipc:endpoint ImageDecoderClient = 7002
These endpoint IDs are duplicated:
85
```
In other news: Our ID space is … interesting :)

### CODE REVIEW

File: Documentation/UsingQtCreator.md [hunk 1]
```diff
@@ -16,3 +18,28 @@ First, make sure you have a working toolchain and can build and run SerenityOS.
...
+- Select the tool "ClangFormat" if not already selected, and click "OK"
+
+Note that not the entire project is clang-format-clean (yet), so sometimes you will see large diffs.
+Use your own judgement whether you want to include such changes. Generally speaking, if it's a few lines then it's a good idea; if it's the entire file then maybe there's a better way to do it, like doing a separate commit, or just ignoring the clang-format changes.
+
+You may want to read up what `git add -p` does (or `git checkout -p`, to undo).
+
+## Compiler Kits
+
+You can slightly improve how well Qt interpretes the code by adding and setting an appropriate "compiler kit".
```
Comment (vkoskiv) [inline Documentation/UsingQtCreator.md:44]: interprets*

setting up* (just a matter of taste, I think!)

Comment (BenWiederhake) [reply Documentation/UsingQtCreator.md:44]: Sure thing – something failed anyway.

Comment (BenWiederhake) [reply Documentation/UsingQtCreator.md]: All done – including the silly shellcheck bug ^^

**Comment by awesomekling** [comment]
Very nice :)

**Pull request merged**